Replace tempdir library with tempfile

ameliabradley opened the pull request:
Replace tempdir library with tempfile:3 because tempdir is no longer
maintained. See advisory: https://rustsec.org/advisories/RUSTSEC-2018-0017

Also move the library to be a dev dependency, as it is only ever used for unit
testing purposes.

Signed-off-by: Amelia Bradley <bradley@bitwise.io>

vaporos commented on libcylinder/src/key/load.rs:
```diff
@@ -246,7 +246,7 @@ mod tests {
     #[test]
     fn load_key_success() {
-        let temp_dir = TempDir::new("test_key_dir").expect("Failed to create temp dir");
+        let temp_dir = TempDir::new().expect("Failed to create temp dir");
```
In order to retain the prefix, this should be Builder::new()::prefix("test_key_dir").tempdir().expect("Failed to create temp dir"); (with a use tempfile::Builder above).

Commit:
Replace tempdir library with tempfile

Replace tempdir library with tempfile:3 because tempdir is no longer
maintained. See advisory: https://rustsec.org/advisories/RUSTSEC-2018-0017

Also move the library to be a dev dependency, as it is only ever used for unit
testing purposes.

Signed-off-by: Amelia Bradley <bradley@bitwise.io>
